cli/src/main/java/com/devonfw/tools/ide/tool/aws/Aws.java

```diff
@@ -0,0 +1,97 @@
+package com.devonfw.tools.ide.tool.aws;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.PosixFilePermission;
+import java.util.Set;
+
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.environment.EnvironmentVariables;
+import com.devonfw.tools.ide.environment.EnvironmentVariablesType;
+import com.devonfw.tools.ide.process.ProcessContext;
+import com.devonfw.tools.ide.tool.LocalToolCommandlet;
+
+/**
+ * {@link LocalToolCommandlet} for AWS CLI (aws).
+ *
+ * @see <a href="https://docs.aws.amazon.com/cli/">AWS CLI homepage</a>
+ */
+
+public class Aws extends LocalToolCommandlet {
+
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   */
+  public Aws(IdeContext context) {
+
+    super(context, "aws", Set.of(TAG_CLOUD));
+  }
+
+  private void makeExecutable(Path file) {
+
+    // TODO this can be removed if issue #132 is fixed
+    Set<PosixFilePermission> permissions = null;
+    try {
+      permissions = Files.getPosixFilePermissions(file);
+      permissions.add(PosixFilePermission.GROUP_EXECUTE);
+      permissions.add(PosixFilePermission.OWNER_EXECUTE);
+      permissions.add(PosixFilePermission.OTHERS_EXECUTE);
+      Files.setPosixFilePermissions(file, permissions);
+    } catch (IOException e) {
+      throw new RuntimeException("Adding execution permission for Group, Owner and Others did not work for " + file, e);
+    }
+  }
+
+  @Override
+  protected void moveAndProcessExtraction(Path from, Path to) {
+
+    if (!this.context.getSystemInfo().isLinux()) {
+      this.context.getFileAccess().move(from, to);
+    } else {
+      // make binary executable using java nio because unpacking didn't preserve the file permissions
+      Path awsInDistPath = from.resolve("dist").resolve("aws");
+      Path awsCompleterInDistPath = from.resolve("dist").resolve("aws_completer");
+
+      // TODO this can be removed if issue #132 is fixed
+      makeExecutable(awsInDistPath);
+      makeExecutable(awsCompleterInDistPath);
+      // running the install-script that aws shipped
+      ProcessContext pc = this.context.newProcess();
+      Path linuxInstallScript = from.resolve("install");
+      pc.executable("bash");
```

Comment:
There's no need to add `bash` as the executable to the processContext, you can directly pass the script as the executable. The processContext checks whether the passed executable is a script, if so it looks for git bash to run it there, see [processContext](https://github.com/devonfw/IDEasy/blob/489c8d6e6775d1f7be3e4b7dc591c12896f074d9/cli/src/main/java/com/devonfw/tools/ide/process/ProcessContextImpl.java#L100-L111)

Reply:
Ah, thanks. I implemented it.

Reply:
Nice :) 
I can't resolve this conversation, could you do it?